lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,146 @@
+      Color indicatorLineColor,
+      Color confirmButtonBackgroundColor,
+      Color cancelButtonBackgroundColor,
+      EdgeInsets indicatorLineEdgePadding,
+      Radius cornerRadius,
+      Radius indicatorLineRadius,
+      double indicatorLineThickness,
+      double indicatorLineWidth}) {
+    return ActionSheetStyle(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      indicatorLineColor: indicatorLineColor ?? this.indicatorLineColor,
+      confirmButtonBackgroundColor:
+          confirmButtonBackgroundColor ?? this.confirmButtonBackgroundColor,
+      cancelButtonBackgroundColor:
+          cancelButtonBackgroundColor ?? this.cancelButtonBackgroundColor,
+      indicatorLineEdgePadding:
+          indicatorLineEdgePadding ?? this.indicatorLineEdgePadding,
+      cornerRadius: cornerRadius ?? this.cornerRadius,
+      indicatorLineRadius: indicatorLineRadius ?? this.indicatorLineRadius,
+      indicatorLineThickness:
+          indicatorLineThickness ?? this.indicatorLineThickness,
+      indicatorLineWidth: indicatorLineWidth ?? this.indicatorLineWidth,
+    );
+  }
+}
+
+class ActionSheet extends StatelessWidget {
+  final ActionSheetViewModel _viewModel;
+  final ActionSheetStyle _style;
+  final ActionSheetListItemStyle _cellStyle;
+
+  const ActionSheet(
+      {Key key,
+      ActionSheetViewModel viewModel,
+      ActionSheetStyle style,
+      ActionSheetListItemStyle cellStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        this._cellStyle = cellStyle,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        color: _style.cornersColor,
+        child: Container(
+            decoration: BoxDecoration(
+                color: _style.backgroundColor,
+                borderRadius: BorderRadius.only(
+                    topLeft: _style.cornerRadius,
+                    topRight: _style.cornerRadius)),
+            child: HeaderAndFooterListView.builder(
+                physics: NeverScrollableScrollPhysics(),
+                shrinkWrap: true,
+                itemCount: _viewModel.actions.length,
+                itemBuilder: (BuildContext context, int index) =>
+                    ActionSheetListItem(
+                        viewModel: ActionSheetListItemViewModel(
+                            _viewModel.actions[index].title,
+                            _viewModel.actions[index].action,
+                            _viewModel.actions[index].isDestructive,
+                            icon: _viewModel.actions[index].icon,
+                            checkBoxIcon:
+                                _viewModel.actions[index].checkBoxIcon),
+                        style: _cellStyle,
+                        numberOfActions: _viewModel.actions.length,
+                        currentAction: index),
+                header: _buildIndicatorLine(_style),
+                footer: RoundedButton.build(
```

Comment:
What about calling the action functions?
I´m expecting to see 2 modes.
1. If the confirm button title is present, then we´re in "Confirm mode", i.e an item is selected when tapped, and then the action of that item is called when the main confirm button is pressed.
2. If there is no confirm button title, then were in "direct mode", the action function is called directly when the action item is pressed.

Reply:
@DavidFD005 could you have pair programming session to look at this. If a confirm button is present tapping an action should select it, and to run its action the user must tab the confirm button. If no confirm button then it should just run its action directly and not select.